src/legacy/core_plugins/kibana/common/field_formats/types/truncate.ts

```diff
@@ -17,26 +17,31 @@
  * under the License.
  */
 
-import _ from 'lodash';
+import { trunc } from 'lodash';
+import { FieldFormat, TEXT_CONTEXT_TYPE } from '../../../../../../plugins/data/common/';
 
 const omission = '...';
 
-export function createTruncateFormat(FieldFormat) {
-  return class TruncateFormat extends FieldFormat {
-    _convert(val) {
-      const length = this.param('fieldLength');
-      if (length > 0) {
-        return _.trunc(val, {
-          'length': length + omission.length,
-          'omission': omission
-        });
-      }
-
-      return val;
-    }
-
+export function createTruncateFormat() {
+  class TruncateFormat extends FieldFormat {
     static id = 'truncate';
     static title = 'Truncated String';
     static fieldType = ['string'];
-  };
+
+    _convert = {
+      [TEXT_CONTEXT_TYPE](this: TruncateFormat, val: any) {
```

Comment:
are you sure we need to specify 'this' type here?

Reply:
Yes, we should define `this` explicitly, else it's looking for inside of the object.